presto-client/src/client.ts

```diff
@@ -0,0 +1,118 @@
+import { PrestoClientConfig, PrestoQuery, PrestoResponse } from './client.types'
+
+export class PrestoClient {
+  private baseUrl: string
+  private catalog?: string
+  private headers: Record<string, string>
+  private interval: number
+  private schema?: string
+  private timezone?: string
+  private user: string
+
+  constructor({ catalog, host, interval, port, schema, timezone, user }: PrestoClientConfig) {
+    this.baseUrl = `${host || 'http://localhost'}:${port || 8080}/v1/statement`
+    this.catalog = catalog
+    this.interval = interval || 100
+    this.schema = schema
+    this.timezone = timezone
+    this.user = user
+
+    this.headers = { 'X-Presto-Client-Info': 'presto-js-client' }
+    if (this.user) {
+      this.headers['X-Presto-User'] = this.user
+    }
+    if (this.timezone) {
+      this.headers['X-Presto-Time-Zone'] = this.timezone
+    }
+
+    // TODO: Set up auth
+  }
+
+  private request({
+    body,
+    headers,
+    method,
+    url,
+  }: {
+    body?: string
+    headers?: Record<string, string>
+    method: string
+    url: string
+  }) {
+    return fetch(url, {
+      body,
+      headers,
+      method,
+    })
+  }
+
+  async query(
+    query: string,
+    options?: {
+      catalog?: string
+      schema?: string
+    },
+  ): Promise<PrestoQuery> {
+    const catalog = options?.catalog || this.catalog
+    const schema = options?.schema || this.schema
+
+    if (!catalog || !schema) {
```

Comment:
even without catalog and schema, a query is still runable if the `<catalog>.<schema>.<table>` format is used. I guess you can leave this to Presto server, it will return an error either the catalog and schema are not in the headers or the query doesn't use fully qualified table names.

Reply:
oh, I hadn't understand that
I'll fix it soon

the other thing I'm missing is the default import behavior, which I'm trying to figure out today